### DISCUSSION TITLE
custom handlebar helper file

### DISCUSSION THREAD

**Discussion by AgentSells** [farling42/obsidian-import-json#67] [Q&A] [answered]
Hi there, 

I really appreciate this obsidian plug-in. And I am trying to leverage it as best I can. I have written a custom handlebar helper file for importing data from a DNDBeyond Character sheet json, to the frontmatter of an Obsidian note. 

I have tested it on handlebarsjs.com and it parses the data perfectly, my helpers are calculating stats HP AC and features etc fine.

But when I try to use the helpers from a .js I get an error "handlebars not defined" I'm sure I am missing something in the js, but googling it has been fruitless.

Is there a tag or reference I need to add or something else to define handlebars?

**Comment by farling42** [comment]
In the js file that you are choosing for the import, are you calling `handlebars.registerHelper` from the main part of the file, and not within a function or class?

Would it be possible to provide a copy of the file, and an example of the hbs file that calls the helper?

**Reply by AgentSells** [reply]
yeah i am

here is an example of exactly how I format a helper

function hb_loud(aString){
	return aString.toUpperCase()
}
Handlebars.registerHelper('loud', hb_loud);

**Reply by AgentSells** [reply, marked as answer]
ooh it is the "H"

handlebarsjs.com uses upper case H in Handlebars.registerHelper, and i copied that, after changing to handlebars.registerHelper, it worked

**Reply by farling42** [reply]
The H was the first thing I noticed 😂

**Reply by AgentSells** [reply]
happy to share the helper file once I have finished formatting some templates to use it but I am really happy that it seems to capture nearly everything from DND Beyond and calculate various things HP and AC.

**Reply by farling42** [reply]
The folks on the Obsidian TTRPG discord might be interested.

No need to share here if it is all working now.

**Reply by AgentSells** [reply]
yeah I'm just testing and mapping to a couple of popular templates, once that's done, I'm in their discord and will share there.
